business/apps.go

```diff
@@ -200,22 +200,17 @@ func fetchNamespaceApps(k8s kubernetes.IstioClientInterface, namespace string, a
 
 // fillCustomDashboardRefs finds all dashboard IDs and Titles associated to this app and add them to the model
 func (in *AppService) fillCustomDashboardRefs(namespace string, app *models.App, details *appDetails) {
-	uniqueRefs := make(map[string]string)
+	allPods := models.Pods{}
 	for _, workload := range details.Workloads {
-		for _, pod := range workload.Pods {
-			for _, ref := range pod.CustomDashboards {
-				if ref != "" {
-					uniqueRefs[ref] = ref
-				}
-			}
-		}
+		allPods = append(allPods, workload.Pods...)
 	}
+	uniqueRefsList := getUniqueRuntimes(allPods)
 	mon, err := kubernetes.NewKialiMonitoringClient()
```

Comment:
If we create a client on every call, perhaps we are hitting some performance issues as well.
I think that in the business layer we can re-use some of existing clients.

Reply:
iirc in the business layer we also create new clients on every calls, or did it change?